stdlib/FileWatching/src/FileWatching.jl

```diff
@@ -232,6 +235,10 @@ mutable struct _FDWatcher
     @static if Sys.iswindows()
         _FDWatcher(fd::RawFD, mask::FDEvent) = _FDWatcher(fd, mask.readable, mask.writable)
         function _FDWatcher(fd::RawFD, readable::Bool, writable::Bool)
+            if fd == RawFD(-1)
+                throw(ArgumentError("Passed file descriptor is $(fd) == RawFD(-1), this is probably not a valid file descriptor"))
+            end
```

Comment:
```suggestion
            fdnum = Core.Intrinsics.bitcast(Int32, fd) + 1
            if fdnum <= 0
                throw(ArgumentError("file descriptor fd=$fd is not a valid file descriptor"))
            end
```

Reply:
Fixed in ed06cc419c, but why do we add 1 to the value? If I understand the previous constructor correctly it's because we use `fdnum` as an index, but that's not done here.

Reply:
Just to avoid a spurious difference with the unix code